lib/DataFileUtil/DataFileUtilImpl.py

```diff
@@ -159,17 +159,21 @@ def _pack(self, file_path, pack):
         self.log('Packing {} to {}'.format(d, pack))
         # tar is smart enough to not pack its own archive file into the new archive, zip isn't.
         # TODO is there a designated temp files dir in the scratch space? Nope.
+        # check dir to archive is not self.tmp or its parent dir for zip
         (fd, tf) = tempfile.mkstemp(dir=self.tmp)
         os.close(fd)
         if pack == 'targz':
             ctf = shutil.make_archive(tf, 'gztar', d)
             suffix = ctf.replace(tf, '', 1)
             shutil.move(ctf, file_path + suffix)
         else:
-            suffix  = '.zip'
+            if os.path.commonprefix([d, self.tmp]) == d:
```

Comment:
I'm not sure I understand this logic. The target zip file will always be a temporary file located in /kb/module/work/tmp/[some uuid]. That means that zipping /kb/module/work/tmp or /kb/module/work/tmp/[some uuid] will be an error.

IIUC, this logic will catch the former case, but not the latter.

Reply:
@MrCreosote This will catch target directory is self.tmp or parent directory of self.tmp. 

The return of os.path.commonprefix([self.tmp, self.tmp]) will be self.tmp.

Reply:
Yep, you're right, duh.

Reply:
Can you add a test for the latter case, actually?

Reply:
@MrCreosote Sounds good. Added test.